Quintet-iOS/View/PickerView.swift

```diff
@@ -125,6 +125,123 @@ struct MonthPicker: View {
+        }
+    }
+}
+
+struct CalendarYearPicker: View {
+    @StateObject var viewModel: DateViewModel
+    @Binding var selectedYear: Int
+
+    let currentYear: Int = Calendar.current.component(.year, from: Date())
+
+    var body: some View {
+        Picker("년도 선택", selection: $selectedYear){
+            ForEach(2017...currentYear, id: \.self) { year in
+                Text("\(Utilities.formatNum(year))년")
+            }
+        }
+        .pickerStyle(WheelPickerStyle())
+    }
+}
+
+struct CalendarYearMonthPickerPopup: View {
+    @ObservedObject var viewModel: DateViewModel
+    @Binding var isShowPopup: Bool
+    
+    init(viewModel: DateViewModel, isShowPopup: Binding<Bool>) {
+        self.viewModel = viewModel
+        _isShowPopup = isShowPopup
+    }
+    
+    var body: some View {
+        GeometryReader{ geometry in
+            VStack{
+                HStack(spacing: -7) {
+                    CalendarYearPicker(viewModel: viewModel, selectedYear: $viewModel.selectedYear)
+                    CalendarMonthPicker(selectedMonth: $viewModel.selectedMonth, viewModel: viewModel)
+                }
+            }
+            .padding(.top, 60)
+            HStack(spacing: 10){
+                Button(action: {
+                    isShowPopup = false
+                    viewModel.updateCalendar()
+                }) {
+                    RoundedRectangle(cornerRadius: 10)
+                        .fill(Color("Background"))
+                        .frame(width: 100, height: 40)
+                        .overlay(
+                            Text("취소")
+                                .foregroundColor(.black)
+                                .font(.system(size: 15))
+                        )
+                }
+                Button(action: {
+                    isShowPopup = false
+                    viewModel.selectedYear = viewModel.selectedYear
+                    viewModel.selectedMonth = viewModel.selectedMonth
+                    viewModel.updateCalendar()
+                }) {
+                    RoundedRectangle(cornerRadius: 10)
+                        .fill(Color("DarkQ"))
+                        .frame(width: 100, height: 40)
+                        .overlay(
+                            Text("확인")
+                                .foregroundColor(.white)
+                                .font(.system(size: 15))
+                        )
+                    
+                }
+            }
+            .padding(.top, 300)
+            .padding(.horizontal, 30)
+            
+        }
+        .padding()
+        .frame(width: 300, height: 400)
+        .background(Color.white)
+        .cornerRadius(30)
+        
+    }
+ }
+
+extension Date {
+    func getAllDates() -> [Date] {
+        let calendar = Calendar.current
+        let startDate = calendar.date(from: Calendar.current.dateComponents([.year, .month], from: self))!
+        let range = calendar.range(of: .day, in: .month, for: startDate)!
```

Comment:
확실하게 nil 값이 들어있지 않아도 forced unwrapping을 활용은 지양하는 것이 좋다고 생각합니다.
습관이 되면 추후에 치명적인 오류를 불러올 수도 있기 때문입니다!
아래와 같이 optional binding을 이용하여 optional 값을 처리하는건 어떨까요?
```suggestion
    guard
        let startDate = calendar.date(
            from: Calendar.current.dateComponents([.year, .month], from: self)
        ),
        let range = calendar.range(of: .day, in: .month, for: startDate)
    else {
        return []
    }
```